### PR TITLE
Increase System.Data.Common test coverage

### PR DESCRIPTION
**Pull request by bgrainger**
For #16337.

Opening a small PR to get feedback on the work so far and check if it meets your standards before submitting more. CC @divega 

### Before

Module | Line | Branch | Method
--- | --- | --- | ---
System.Data.Common | 55.03% | 51.8% | 60.89%

### After

Module | Line | Branch | Method
--- | --- | --- | ---
System.Data.Common | 59.27% | 54.83% | 63.15%

### CODE REVIEW

**Comment by divega** [comment]
cc @roji, @ajcvickers

**Comment by bgrainger** [comment]
BTW, are code coverage reports working? I can't access ci.dot.net, as [described here](https://github.com/dotnet/corefx/blob/master/Documentation/building/code-coverage.md).

I mentioned this in #38191 but that part of my bug report wasn't specifically addressed.

**Comment by danmosemsft** [comment]
@ViktorHofer the code coverage docs still talk about Jenkins. Are there regular accessble runs somewhere else.

**Comment by ViktorHofer** [comment]
We don't have regular corefx code coverage runs as of speaking. The remaining work is tracked here: https://github.com/dotnet/corefx/issues/34886. I'll update the docs meanwhile.

**Comment by danmosemsft** [comment]
Thanks @ViktorHofer

**Comment by danmosemsft** [comment]
ping @roji

File: src/System.Data.Common/tests/System.Data.Common.Tests.csproj [hunk 1]
```diff
@@ -16,6 +16,7 @@
     <Compile Include="System\Data\Common\DataAdapterTest.cs" />
     <Compile Include="System\Data\Common\DataColumnMappingCollectionTest.cs" />
     <Compile Include="System\Data\Common\DataTableMappingCollectionTest.cs" />
+    <Compile Include="System\Data\Common\DataTableMappingCollectionTest2.cs" />
```
[the remark this answers is not in the document]
Comment (bgrainger) [reply src/System.Data.Common/tests/System.Data.Common.Tests.csproj]: There were existing `XyzTest2.cs` files in the repo, so I copied that "pattern". My primary reason for starting a new file (which I suspect was the genesis of the other "duplicate" files, too) was avoiding the `// (C) Ameya Gargesh`, `Copyright (C) 2004 Novell, Inc`, and inapplicable license statements at the top of the existing files.

**Comment by bgrainger** [comment]
> Everything looks good, I learned about some ancient ADO.NET types I didn't even knew existed :)

Right?!

File: src/System.Data.Common/tests/System.Data.Common.Tests.csproj (hunk 1 above)
[the remark this answers is not in the document]
Comment (roji) [reply src/System.Data.Common/tests/System.Data.Common.Tests.csproj]: Makes sense.

**Comment by roji** [comment]
Yep, I think we can say you're now an ADO.NET archeologist...

**Pull request merged**